### ISSUE TITLE
Accessibility Testing for Accredited Representation Management, Appoint a Representative MVP

### ISSUE THREAD

**Issue by oddball-lindsay** [department-of-veterans-affairs/va.gov-team#97463]
Platform guidance: https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review

### Product information

- [x] Team name, product name, and feature name have been added to the title of this issue.
- [x] Team label, product label and feature label (if applicable) have been added to this issue.

### Who completed the use of color and color contrast test?

@jquispe-oddball tested with the Chrome browser for both desktop and mobile (Android).
@oddball-lindsay tested with the Safari browser for both desktop (Macbook Air) and mobile (iPhone 13 Pro). 

### Use of color and color contrast checks
Tools used: Wave and Axe extensions for Chrome, Colour Contrast Analyzer for Mac.

- [x] All text of 20px or smaller has a 4.5:1 contrast ratio to its background (or better)
- [x] All text of 20px or larger has a 3:1 contrast ratio to its background (or better)
- [x] Non-text elements have a 3:1  contrast ratio to their background and to neighboring elements (or better)
- [x] Color is not the only way to distinguish links from other text (eg. links are underlined)
- [x] Any charts, maps, infographics, and tables convey all information without only relying on color
- [x] Content does not refer to color, especially when providing user instructions (eg. "Click the blue button")

### How did color testing go?

Chrome: No color contrast issues. WAVE tool did flag two errors per page related to color contrast, but it is not telling me which element it is. I believe this is an error related to WAVE. AXE did not flag any color contrast issues

Safari: No color contrast issues.

### Who completed the axe scans?

@jquispe-oddball 

### axe checks

- [x] Each page has been scanned using axe (results shared below)
- [x] axe is integrated into your end-to-end testing

### How did axe testing go?

- There is one critical error. On representative-search, the input field has no label. 
  - This will be addressed in https://github.com/department-of-veterans-affairs/va.gov-team/issues/97039 before release.
- There is a moderate error on some pages where the heading order is incorrect. 
  - These are captured in this [parent ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/98024), and we have yet to determine if these will be addressed prior to release.
- There are some flags related to “best practices” but it is not a blocker. Some radio input fields are being flagged as having an incorrect role.

More details, including screenshots, in the comment below.

### Who completed the content zoom and reflow test?

@jquispe-oddball tested with the Chrome browser for both desktop and mobile (Android).
@oddball-lindsay tested with the Safari browser for both desktop (Macbook Air) and mobile (iPhone 13 Pro).

### Content zoom and reflow checks (desktop AND mobile)

**Desktop**:
- [x] All page elements are readable and usable at 200% zoom
- [x] All page elements are readable and usable at 300% zoom
- [x] All page elements are readable and usable at 400% zoom

Successfully tested on desktop (Chrome and Safari) and all page elements are readable and usable at 200, 300, and 400% zoom.

**Mobile**: 

- [ ] All page elements are readable and usable at 200% zoom
- [ ] All page elements are readable and usable at 300% zoom
- [ ] ~All page elements are readable and usable at 400% zoom~ Chrome on Android and Safari on iPhone both have a limit of 300%. 

For both mobile browsers, many page elements are not readable and usable at 200% and 300% zoom. This includes:
- Issues with text not wrapping / getting cut off
- Issues with resizing
- Issues with content taking up the entire screen (Safari on iPhone at 300%)

More details, including screenshots, in the comments below.

### Who completed the keyboard navigation test?

@jquispe-oddball tested with the Chrome browser.
@oddball-lindsay tested with the Safari browser.

### Keyboard navigation checks

- [ ] Each link, button, form input, checkbox, radio button, select menu, and custom element can receive keyboard focus
- [ ] Each link, button, form input, checkbox, radio button, select menu, and custom element responds to expected keys
- [ ] All elements under focus have a visible focus indicator
- [ ] The order of [Tab] stops made sense and was appropriate for completing tasks

### How did keyboard testing go?

Chrome:
Each link, button, and form elements like checkbox, radio, and select had a focus indicator. I was able to navigate radio buttons with the arrow keys as well.
Tab order was fine as well, no major issues to report.

Safari is missing focus for the majority of links, buttons, form inputs, etc.
- `/introduction` page: I could only tab to / place focus on the "If you don't know who you'd like to appoint" additional component.
- `/claimant-type` page: I couldn't tab to / place focus on any elements.
- `representative-select` page:  I could only tab to / place focus on the search bar.
- Veteran chapter: I could only tab to / place focus on the text input fields.
- Authorization chapter + `review-and-submit` and `download` pages: I couldn't tab to / place focus on any elements.
- `next-steps` page: I could only tab to / place focus on the representative address, in the lower contact card.

### Do you have any other results to share?

- Sometimes when I go into a new page, the page loads and I can only see the footer. I have to scroll up to see the content. I can replicate this while navigating with keyboard or mouse.
  - This will be addressed in https://github.com/department-of-veterans-affairs/va.gov-team/issues/95661 before release.
- If I am on any step in the form, and hit refresh, it returns an error page.
  - This is captured here https://github.com/department-of-veterans-affairs/va.gov-team/issues/98430, and we have yet to determine if these will be addressed prior to release.

**Comment by jquispe-oddball** [comment]
I tested with WAVE and Axe.

**How did color testing go?**
No color contrast issues. WAVE tool did flag two errors per page related to color contrast, but it is not telling me which element it is. I believe this is an error related to WAVE. AXE did not flag any color contrast issues

**How did axe testing go?**
There is one critical error. On representative-search, the input field has no label. This might be logged in the google documentation.
![Image](https://github.com/user-attachments/assets/d637a96e-ea0a-4628-9bdb-c57195e148bc)

There is a moderate error on some pages where the heading order is incorrect. These are logged in the google sheet.
![Image](https://github.com/user-attachments/assets/489f0a73-5ebc-4875-aa04-7956c1d76b17)

Same error was logged using WAVE:
![Image](https://github.com/user-attachments/assets/3877df26-095a-499e-9ec5-c4fc943eb554)

There are some flags related to “best practices” but it is not a blocker. Some radio input fields are being flagged as having an incorrect role. 


**How did content zoom and reflow testing go?**
I was able to successfully test on desktop at 200, 300, and 400% zoom. Android chrome has a limit of 300%.
At 200% and 300% zoom on mobile, it is not usable. I have attached screenshots that show the text being clipped completely. Sometimes the text does resize properly, but you will notice some clipping on almost every form page. 

On representative search, if I am on 300% zoom, I cannot see what I typed in.
![Image](https://github.com/user-attachments/assets/7e2c8d88-213f-436b-96f4-6d1aaed624e6)
![Image](https://github.com/user-attachments/assets/e2145f80-ed55-4759-b5b9-5a7880fb1e9a)
![Image](https://github.com/user-attachments/assets/89d7b770-66bb-4969-aee5-7a8dfde3f8fc)
![Image](https://github.com/user-attachments/assets/7216c24c-cae7-4a33-9f3f-5b487231e6d9)


**How did keyboard testing go?**
Each link, button, and form elements like checkbox, radio, and select had a focus indicator. I was able to navigate radio buttons with the arrow keys as well.
Tab order was fine as well, no major issues to report.


**Do you have any other results to share?**
Sometimes when I go into a new page, the page loads and I can only see the footer. I have to scroll up to see the content. I can replicate this while navigating with keyboard or mouse. 
If I am on any step in the form, and hit refresh, it returns an error page.

**Comment by oddball-lindsay** [comment]
**Draft comment**

**Comment by oddball-lindsay** [comment]
- [x] Color contrast testing
- [ ] Zoom testing
- [ ] Keyboard navigation testing

**Comment by oddball-lindsay** [comment]
> If I am on any step in the form, and hit refresh, it returns an error page.

Steps to Reproduce (using Chrome browser):
1. Login as a test user
2. Visit the [Appoint a Rep intro page](https://staging.va.gov/get-help-from-accredited-representative/appoint-rep/)
3. Click the entry point to start the Appoint experience
4. Navigate to partway through the experience -- I stopped on the `representative-replace` screen
5. Click "refresh" on the browser
6. A modal pops up warning that content may be lost -- proceed with the refresh
7. A blank page loads![Image](https://github.com/user-attachments/assets/243fd1e8-73ac-4ee6-b3f9-4dfbe860331e)

The unauthenticated experience reroutes back to the `/introduction` page when a user bypasses the warning modal. Can we do the same for the authenticated experience?

**Comment by oddball-lindsay** [comment]
Weird things I saw on Safari (iPhone) at 200% and 300% zoom.

### 200% zoom

The word "representation" is too long and gets cut off:
![Image](https://github.com/user-attachments/assets/9cd827fe-900e-4791-876a-968f8dde7051)

Text is not wrapping:
![Image](https://github.com/user-attachments/assets/5488d5d9-11a8-47cf-9c82-5dfd9c918ce7)

The search box is hard to see and both the email and button are not resizing/wrapping within their container: 
![Image](https://github.com/user-attachments/assets/58b8b46a-c524-4690-83a1-305d1232bd91)


### 300% zoom

Same issues as the 200% zoom, with the added issue where now the content only takes up half of the visible area:
![Image](https://github.com/user-attachments/assets/163872ba-ad34-490f-b42b-f8fc36998570)
![Image](https://github.com/user-attachments/assets/c312c089-e03f-4588-9fce-4128b1bf8af8)

**Comment by oddball-lindsay** [comment]
@shiragoodman @briandeconinck this is the first time we've had findings for this Accessibility Testing, prior to Staging Review. We have found issues with zoom on mobile browsers, as well as issues with focus on the Safari browser.

Should I keep this ticket open? Or close it since our testing is complete, and any accessibility issues can be tackled in other tickets?

**Comment by shiragoodman** [comment]
@oddball-lindsay we really don't have a preference for whether you keep it open or not.. however @briandeconinck can correct me if he does have a preference. As long as it's available to us, and linked in your SR artifacts section, we should be ok.

**Comment by oddball-lindsay** [comment]
Thanks @shiragoodman - this is indeed linked in the AR artifacts, so I'll close this out for now. 

@briandeconinck if you have a different preference, just let me know and I can adjust accordingly!
